### DISCUSSION TITLE
Map should have more contrast

### DISCUSSION THREAD

**Discussion by mltiede** [home-assistant/frontend#12932] [Dashboard] [answered]
I have a person badge on the dashboard.  When I click on it, it takes me to a map showing where the person is.  But the map, at least in dark mode on my Galaxy S10, is too dark to make out where they are on the map.  I can see hints of roads that are just a dark gray on a what looks like a black background.  And the screen captures on Phone Link on the PC that I attached here look better than they do on the phone.

![darkmodemap](https://user-images.githubusercontent.com/104257005/173193418-bb7dcb80-2e47-4b65-b024-bffdbe64bc7f.png)

In light mode, the roads show up a little better, but I see no reason to have such low contrast on the items on the map.

![lightmodemap](https://user-images.githubusercontent.com/104257005/173193320-884899df-8680-49a6-8c7f-180d4e8a54d4.png)

**Comment by ildar170975** [comment]
Agree. Map is too hard to read in a dark mode on PC too.
![image](https://user-images.githubusercontent.com/71872483/173205555-fca675ce-a993-4288-ab8d-bf02b4b77159.png)

In a light mode it is a bit better:
![image](https://user-images.githubusercontent.com/71872483/173205600-4e47e672-ce3a-45a3-bd55-d228f576fb3a.png)

Could it be because of a map provider?

**Comment by KTibow** [comment]
We currently use CARTO. Examples:
Light "Positron": https://d.basemaps.cartocdn.com/light_all/7/25/46@2x.png
![](https://d.basemaps.cartocdn.com/light_all/7/25/46@2x.png)
Dark "Dark Matter": https://d.basemaps.cartocdn.com/dark_all/7/25/46@2x.png
![](https://d.basemaps.cartocdn.com/dark_all/7/25/46@2x.png)

They also have a version we don't use,
Color "Voyager": https://d.basemaps.cartocdn.com/rastertiles/voyager/7/25/46@2x.png
![](https://d.basemaps.cartocdn.com/rastertiles/voyager/7/25/46@2x.png)
And a third party's satellite view:
https://1.aerial.maps.api.here.com/maptile/2.1/maptile/newest/satellite.day/7/25/46/256/jpg?lg=eng&token=0&app_id=0
![](https://1.aerial.maps.api.here.com/maptile/2.1/maptile/newest/satellite.day/7/25/46/256/jpg?lg=eng&token=0&app_id=0)

I would recommend applying `filter: brightness(4) contrast(1.33)` to the dark map in order to give it more contrast, like this:
![image](https://user-images.githubusercontent.com/10727862/179019384-5b28e00f-fc26-463b-8b90-c831322c1829.png)

**Reply by mltiede** [reply]
So how do you apply a filter in home assistant?  I don't see any place to change settings for map.

**Reply by ildar170975** [reply]
This may be done by using either card-mod for a particular Map card or a custom card-mod-theme to style all Map cards.

**Reply by mltiede** [reply]
I'm not using a card. I was just tracking a person. When I click on the person badge, the map is shown.

**Reply by ildar170975** [reply]
> I'm not using a card. I was just tracking a person. When I click on the person badge, the map is shown.

To apply a filter, you need to create a custom card-mod theme and specify a style for a "map" element inside a more-info window.

**Reply by mltiede** [reply]
How do I get that to show up for the badge?

**Reply by KTibow** [reply]
#13200

**Comment by matthiasdebaat** [comment, marked as answer]
_Edit: Fixed in https://github.com/home-assistant/frontend/pull/13227_

Personally, I would like to use a more vibrant map style with more contrast that users are familiar with, for example from Google Maps. For the CARTO basemap, that would be the [Voyager style](https://d.basemaps.cartocdn.com/#13/40.7062/-74.0045). We're only adding zones and people to our map, so I think this would work perfectly with a more detailed map.

Downside is that there isn't a dark mode version of Voyager. In that case we could invert the colors or apply a hue rotate effect, [described in this blog post](https://www.esri.com/arcgis-blog/products/arcgis-online/mapping/create-a-light-or-dark-version-of-any-map-in-two-seconds/).

<img width="846" alt="CleanShot 2022-07-18 at 11 16 28@2x" src="https://user-images.githubusercontent.com/20203284/179481078-7cce2304-6f37-472f-8e34-9790bcd43cda.png">
<img width="741" alt="CleanShot 2022-07-18 at 11 17 35@2x" src="https://user-images.githubusercontent.com/20203284/179481272-8e532b06-eb24-4790-8df4-c8e221e353af.png">

**Reply by KTibow** [reply]
How's `invert(1) hue-rotate(180deg) brightness(3) contrast(1.3) saturate(0.5)` for dark mode?
![image](https://user-images.githubusercontent.com/10727862/179773625-64269ec5-919e-4866-8505-9e9215b86cb5.png)
